circe/src/main/scala/com/softwaremill/sttp/circe.scala

```diff
@@ -0,0 +1,32 @@
+package com.softwaremill.sttp
+
+import com.softwaremill.sttp.model.{
+  RequestBody,
+  ResponseAs,
+  SerializableBody,
+  StringBody
+}
+import io.circe.parser._
+import io.circe.{Decoder, Encoder}
+
+import scala.language.higherKinds
+
+package object circe {
+  private[sttp] val ApplicationJsonContentType = "application/json"
+
+  private def serializeBody[B](body: B)(
+      implicit encoder: Encoder[B]): RequestBody[Nothing] =
+    SerializableBody((b: B) => StringBody(encoder(b).noSpaces, Utf8), body)
+
+  implicit final class CirceRequestTOps[U[_], T, +S](val req: RequestT[U, T, S])
+      extends AnyVal {
+    def jsonBody[B: Encoder](body: B): RequestT[U, T, S] =
```

Comment:
That sounds like it could work. However, since the body is not serialized until the request is actually evaluated, the `defaultContentType` would have to be set on the `SerializableBody`, right?